src/main/java/org/cryptomator/cryptofs/health/dirid/OrphanDir.java

```diff
@@ -134,10 +158,53 @@ Path prepareRecoveryDir(Path pathToVault, FileNameCryptor cryptor) throws IOExce
 		return new CryptoPathMapper.CiphertextDirectory(stepParentUUID, stepParentDir);
 	}
 
+	//visible for testing
+	Optional<String> retrieveDirId(Path orphanedDir, Cryptor cryptor) {
+		var dirIdFile = orphanedDir.resolve(Constants.DIR_ID_FILE);
+		var dirIdBuffer = ByteBuffer.allocate(36); //a dir id contains at most 36 ascii chars, in this impl encoded in utf8
+
+		try (var channel = Files.newByteChannel(dirIdFile, StandardOpenOption.READ); //
+			 var decryptingChannel = wrapDecryptionAround(channel, cryptor)) {
+			decryptingChannel.read(dirIdBuffer);
+			dirIdBuffer.flip();
+		} catch (IOException e) {
+			LOG.info("Unable to read dirIdFile of {}.", orphanedDir, e);
+			return Optional.empty();
+		}
+
+		var allegedDirId = StandardCharsets.UTF_8.decode(dirIdBuffer).toString();
```

Comment:
It definitively increases readability, but what i like about the original approach is its resiliency: By definition the directory id is no longer than 36 ASCII chars (bytes when utf-8 encoded). Even if the dirId file is bloated with a lot of text, we don't care. and at most read 36 bytes and continue.

Reply:
Discussion took place over a different channel, the conclusion is:
* The directory id _is_ defined (see https://docs.cryptomator.org/en/latest/security/architecture/#directory-ids)
* `ReadableByteChannel::read` is problematic, since it might not read all bytes
* cryptolib offers a [utility method](https://github.com/cryptomator/cryptolib/blob/ccfe34dc0052f906ad9c840c73f715e2ae6d4837/src/main/java/org/cryptomator/cryptolib/common/ByteBuffers.java#L46-L55) to read from a channel until either channel ends or buffers capacity is reached

Reply:
Changed in [1d9c9b4](https://github.com/cryptomator/cryptofs/pull/127/commits/1d9c9b436377d42b455f627a709b2d193b17cb72)

I used the utility method from cryptolib.